src/client/metadataApiRetrieve.ts

```diff
@@ -30,15 +30,29 @@ export type MetadataApiRetrieveOptions = MetadataTransferOptions &
 
 export class RetrieveResult implements MetadataTransferResult {
   public readonly response: MetadataApiRetrieveStatus;
+  // The ComponentSet of retrieved components.
   public readonly components: ComponentSet;
 
-  constructor(response: MetadataApiRetrieveStatus, components: ComponentSet) {
+  // The ComponentSet of just the components on the local file system.
+  private localComponents: ComponentSet;
+  private fileResponses: FileResponse[];
+
+  constructor(
+    response: MetadataApiRetrieveStatus,
+    components: ComponentSet,
+    localComponents?: ComponentSet
+  ) {
     this.response = response;
     this.components = components;
+    this.localComponents = new ComponentSet(localComponents?.getSourceComponents());
   }
 
   public getFileResponses(): FileResponse[] {
-    const responses: FileResponse[] = [];
+    if (this.response && this.fileResponses) {
```

Comment:
Why move this up to the instance?  Do some our consumers call `getFileResponses` multiple times and you're trying to save work?

Reply:
Yes, it's called multiple times and there's no reason to do the work more than once after we get the response.